Shared/Account/AccountLoginView.swift

```diff
@@ -0,0 +1,106 @@
+
+    @State private var isShowingAlert: Bool = false
+    @State private var alertMessage: String = ""
+
+    var body: some View {
+        VStack {
+            HStack {
+                Image(systemName: "person.circle")
+                    .foregroundColor(.gray)
+                #if os(iOS)
+                TextField("Username", text: $account.username)
+                    .keyboardType(.emailAddress)
+                    .autocapitalization(.none)
+                    .disableAutocorrection(true)
+                    .textFieldStyle(RoundedBorderTextFieldStyle())
+                #else
+                TextField("Username", text: $account.username)
+                #endif
+            }
+            HStack {
+                Image(systemName: "lock.circle")
+                    .foregroundColor(.gray)
+                #if os(iOS)
+                SecureField("Password", text: $account.password)
+                    .autocapitalization(.none)
+                    .disableAutocorrection(true)
+                    .textFieldStyle(RoundedBorderTextFieldStyle())
+                #else
+                SecureField("Password", text: $account.password)
+                #endif
+            }
+            HStack {
+                Image(systemName: "link.circle")
+                    .foregroundColor(.gray)
+                #if os(iOS)
+                TextField("Server URL", text: $account.server)
+                    .keyboardType(.URL)
+                    .autocapitalization(.none)
+                    .disableAutocorrection(true)
+                    .textFieldStyle(RoundedBorderTextFieldStyle())
+                #else
+                TextField("Server URL", text: $account.server)
+                #endif
+            }
+            Spacer()
+            if account.isLoggingIn {
+                ProgressView("Logging in...")
+                    .padding()
+            } else {
+                Button(action: {
+                    account.login(
+                        to: account.server,
+                        as: account.username, password: account.password,
+                        completion: loginHandler
+                    )
+                }, label: {
+                    Text("Login")
+                })
+                .disabled(account.isLoggedIn)
+                .padding()
+            }
+        }
+        .alert(isPresented: $isShowingAlert) {
+            Alert(
+                title: Text("Error Logging In"),
+                message: Text(alertMessage),
+                dismissButton: .default(Text("OK"))
+            )
+        }
+    }
+
+    func loginHandler(result: Result<UUID, AccountError>) {
+        do {
+            _ = try result.get()
+        } catch AccountError.serverNotFound {
+            alertMessage = """
+The server could not be found. Please check that you've entered the information correctly and try again.
+"""
+            isShowingAlert = true
+        } catch AccountError.invalidCredentials {
```

Comment:
This isn't fully in the API docs, but there are two login errors that can happen:

`401` - invalid password
`404` - user[name] doesn't exist

We make this distinction in the API to give users a better login experience (a forgotten username is a common problem), so we should definitely surface that in the SwiftUI client, too.

(Side note: the `error_msg` field on the login endpoint is a user-friendly message that can optionally be exposed directly to the user, as it describes exactly what went wrong. E.g. a `404` might come with the message: `Please enter your username instead of your email address.`)

Reply:
The Swift package I'm going to work on integrating this week passes through those user-friendly messages, so I'll add these particular cases to the `AccountError` enum to hook things up nicely. 👍